queue_harvesting.py

```diff
@@ -2,257 +2,113 @@
-            message=message,
-            media_type='video',
-            media_savepath=var.MEDIA_SAVEPATH)
+        media_key, provider = message['payload']['key'], message['source']
 
-        # Initialize the time variables.
-        start_time = time.time()
-        total_time_preprocessing = 0
-        total_time_class_prediction = 0
-        total_time_processing = 0
-        total_time_postprocessing = 0
-        start_time_preprocessing = time.time()
-
-        # Perform object classification on the media
-        # initialise the yolo model, additionally use the device parameter to specify the device to run the model on.
-        device = 'cuda' if torch.cuda.is_available() else 'cpu'
-        MODEL = YOLO(var.MODEL_NAME).to(device)
-        MODEL2 = None
-        if var.MODEL_NAME_2:
-            MODEL2 = YOLO(var.MODEL_NAME_2).to(device)
-        if var.LOGGING:
-            print(f'3) Using device: {device}')
-
-        # Open video-capture/recording using the video-path. Throw FileNotFoundError if cap is unable to open.
-        if var.LOGGING:
-            print(f'4) Opening video file: {var.MEDIA_SAVEPATH}')
-        if not os.path.exists(var.MEDIA_SAVEPATH):
-            raise FileNotFoundError(f'Cannot find {var.MEDIA_SAVEPATH}')
-        if not var.MEDIA_SAVEPATH.lower().endswith(('.mp4', '.avi', '.mov')):
-            raise TypeError('Unsupported file format! Only support videos with .mp4, .avi, .mov extensions')
-        cap = cv2.VideoCapture(var.MEDIA_SAVEPATH)
-        if not cap.isOpened():
-            raise FileNotFoundError('Unable to open video file')
-        video_out = None
-        # Initialize the video-writer if the SAVE_VIDEO is set to True.
-        if var.SAVE_VIDEO:
-            fourcc = cv2.VideoWriter.fourcc(*'avc1')
-            video_out = cv2.VideoWriter(
-                filename=var.OUTPUT_MEDIA_SAVEPATH,
-                fourcc=fourcc,
-                fps=var.CLASSIFICATION_FPS,
-                frameSize=(int(cap.get(cv2.CAP_PROP_FRAME_WIDTH)),
-                           int(cap.get(cv2.CAP_PROP_FRAME_HEIGHT)))
-            )
-
-        # Initialize the classification process.
-        # 2 lists are initialized:
-        # Classification objects
-        # Additional list for easy access to the ids.
+        time_verbose = TimeVerbose()
+        cap = harvest_service.open_video(message)
 
         # frame_number -> The current frame number. Depending on the frame_skip_factor this can make jumps.
-        # predicted_frames -> The number of frames, that were used for the prediction. This goes up by one each prediction iteration.
-        # frame_skip_factor is the factor by which the input video frames are skipped.
+        # predicted_frames -> The number of predicted frames that satisfies the condition.
+        # frame_skip_factor -> the factor by which the input video frames are skipped.
         frame_number, predicted_frames = 0, 0
         frame_skip_factor = int(
```

Comment:
It would be nice if we can remove all the "abstracts variables", such as `frame_skip_factor`, `frame_number` in the harvesting class. From a user point of view they should actual not care about this in the code. They are just expected to set the processing frame rate, max number of detections through environment variables.

I would expect there would be a clean function like `frame = harvest.process()` which return a desired frame.